### PR TITLE
Upgrade go version to 1.21.3

### PR DESCRIPTION
**Pull request by raj-prince**
### Description
Upgrading go version from 1.21.2 to 1.21.3

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Automation
3. Integration tests - Successfully passed.

### CODE REVIEW

**Comment by raj-prince** [comment]
Integration test passed successfully.

Perf metrics:

image

File: tools/package_gcsfuse_docker/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ ARG DEBFULLNAME="GCSFuse Team"
 # Build Arg for building through a particular branch/commit. By default, it uses
 # the tag corresponding to passed GCSFUSE VERSION
 ARG BRANCH_NAME="v${GCSFUSE_VERSION}"
-RUN git checkout "${BRANCH_NAME}"
+RUN git checkout "go_1_21_3"
```
Comment (ashmeenkaur) [inline tools/package_gcsfuse_docker/Dockerfile:40]: revert this before merging

File: perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh [hunk 2]
```diff
@@ -88,6 +88,7 @@ if [[ "$perfTestStr" == *"$EXECUTE_PERF_TEST_LABEL"* ]];
 then
  # Executing perf tests for master branch
  install_requirements
+ git reset --hard
```
Comment (ashmeenkaur) [inline perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh:91]: revert this before merging

**Review by ashmeenkaur — APPROVED** [review]
LGTM

**Review by sethiay — APPROVED** [review]

**Pull request merged**